### DISCUSSION TITLE
렉시컬 스코프와 동적 스코프의 차이점은 무엇인가요?

### DISCUSSION THREAD

**Discussion by lledellebell** [diving-javascript/you-dont-know-js-yet#62] [1. 스코프] [answered]
<details>
<summary>관련 챕터</summary>
[Part2] 1.6 렉시컬 스코프 (Lexical Scope)
</details>

**Comment by wogus216** [comment, marked as answer]
<details>
<summary>답변</summary>




### 렉시컬 스코프

코드를 작성할 때 변수와 함수의 유효 범위를 결정함
함수가 정의된 위치에 따라 변수의 유효 범위가 결정,실행시점이 아니라 작성 시점의 코드 구조에 따라 변수의 스코프가 정해짐

### 동적 스코프

함수가 호출될 때의 호출 스택에 따라 변수의 유효범위를 결정함
실행 시점의 함수 호출 스택에 따라 변수의 스코프가 정해짐,JS는 렉시컬 스코프다.


####차이점
1. 스코프 결정 시점: 렉시컬은 코드 작성시점, 동적 스코프는 함수 호출 시점에 결정
2. 변수 참조 방식: 렉시컬은 함수 정의 위치에 따라 외부 변수에 접근, 동적 스코프는 함수 호출 스택에 따라 외부 변수에 접근
</details>
